dotCMS/src/main/java/com/dotcms/rest/api/v1/folder/FolderView.java

```diff
@@ -0,0 +1,102 @@
+package com.dotcms.rest.api.v1.folder;
+
+import com.dotmarketing.portlets.folders.model.Folder;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * This class holds a view with all attributes of com.dotmarketing.portlets.folders.model.Folder
+ * plus a list to hold the children of a folder
+ *
+ *
+ */
+public class FolderView {
+
+    private final String path;
+    private final String defaultFileType;
+    private final String filesMasks;
+    private final Date iDate;
+    private final String hostId;
+    private final String identifier;
+    private final String inode;
+    private final Date modDate;
+    private final String name;
+    private final Boolean showOnMenu;
+    private final Integer sortOrder;
+    private final String title;
+    private final String type;
+    private List<FolderView> subFolders;
```

Comment:
Codacy found an issue: [Private field 'subFolders' could be made final; it is only initialized in the declaration or constructor.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5916479)